packages/gatsby/src/schema/resolvers.ts

```diff
@@ -472,7 +475,19 @@ export const defaultFieldResolver: GatsbyResolver<
   return null
 }
 
-export function tracingResolver<TSource, TArgs>(
+let WARNED_ABOUT_RESOLVERS = false
+function badResolverInvocationMessage(missingVar: string, path?: Path): string {
+  const resolverName = path ? `${pathToArray(path)} ` : ``
+  return `GraphQL Resolver ${resolverName}got called without "${missingVar}" argument. This might cause unexpected errors.
+  
+It's likely that this has happened in a schemaCustomization with manually invoked resolver. If manually invoking resolvers, it's best to invoke them as follows:
+
+  resolve(parent, args, context, info)
```

Comment:
Folks who might be not well versed in graphql stuff might not know where those args (context, info) come from - might be good to include some information that those are args passed to every resolver - I was thinking about either linking to https://graphql.org/learn/execution/#root-fields-resolvers (not quite ideal because it's not just resolver stuff, but it explains resolver args) or copying some information from it?

Not a blocker for this PR tho - I'd say let's ship fix and warning and possibly iterate on it if we get issues/questions about it?

Reply:
Yeah, we need better docs on this. But that needs to happen with the whole schema customization workflow